File: csharp/squeaky-clean/SqueakyClean.cs
using System.Linq;
using System.Text.RegularExpressions;

public static class Identifier
{
    private static bool isGreekLc(char c) => 'α' <= c && c <= 'ω';

    public static string Clean(string identifier) =>
        Regex.Replace(identifier, @"-(\D)", m => m.Groups[1].Value.ToUpper()).Select(c =>
            c switch
            {
                ' ' => "_",
                '\0' => "CTRL",
                _ when isGreekLc(c) => "",
                _ when char.IsLetter(c) => $"{c}",
                _ => ""
            }
        ).Aggregate("", (acc, n) => acc + n);
}

File: java/squeaky-clean/src/main/java/SqueakyClean.java
import java.util.regex.Matcher;
import java.util.regex.Pattern;
import java.util.stream.Collectors;

class SqueakyClean {
    static String clean(String identifier) {

        Matcher matcher = Pattern.compile("-(\\D)").matcher(identifier);

        if (matcher.find()) identifier = matcher.replaceAll(matcher.group(1).toUpperCase());

        return identifier.chars()
                .mapToObj(c -> c == ' ' ? "_" : Character.isISOControl(c) ? "CTRL" :
                        c >= 'α' && c <= 'ω' ? "" : Character.isLetter(c) ?
                                String.valueOf((char) c) : ""
                ).collect(Collectors.joining(""));
    }
}
